### ISSUE TITLE
Datapack functions using entity_type tags don't work after being loaded for the first time

### ISSUE THREAD

**Issue by GoldenDelicios** [PaperSpigot/Paper#7293]
### Expected behavior

Functions in Minecraft datapacks should be able to reference custom entity_type tags in their commands.
e.g. If a function at `data/mydatapack/functions/kill_arthropods.mcfunction` contains
```
kill @e[type=#mydatapack:arthropods]
```
and an entity tag at `data/mydatapack/tags/entity_types/arthropods.json` contains
```json
{
  "replace": false,
  "values": [
    "minecraft:spider",
    "minecraft:cave_spider",
    "minecraft:silverfish",
    "minecraft:endermite",
    "minecraft:bee"
  ]
}
```
and I run `/function mydatapack:kill_arthropods`, all entities of the respective types should be killed.

### Observed/Actual behavior

No entities match the selector `@e[type=#mydatapack:arthropods]` and no entities are killed.
This is not an issue if the command `/kill @e[type=#mydatapack:arthropods]` is run in-game, or after reloading the datapack for the second time.

### Steps/models to reproduce

1. Start the server.
2. Add [mydatapack.zip](https://github.com/PaperMC/Paper/files/7825235/mydatapack.zip) to the datapacks folder.
3. Run `/minecraft:reload` to enable the datapack.
4. Spawn a spider or other mob in `arthropods.json`.
5. Run `/function mydatapack:kill_arthropods`.
6. Observe that the spider does not die.
7. Repeat steps 3-5. Observe that the spider now dies.
8. Restart the server and repeat steps 4 and 5. Observe that the spider does not die.

### Plugin and Datapack List

No plugins, no datapacks other than the test datapack attached above.

### Paper version

This server is running Paper version git-Paper-134 (MC: 1.18.1) (Implementing API version 1.18.1-R0.1-SNAPSHOT) (Git: 55a1756)
You are 2 version(s) behind
Download the new version at: https://papermc.io/downloads
Previous version: git-Paper-398 (MC: 1.17.1)

### Other

My suspicion is that the functions are processed before the entity tags when the datapack is first loaded. I've tested this on a vanilla 1.18.1 server and could not reproduce the issue.

**Comment by Owen1212055** [comment]
Seems to be fixed.

This server is running Paper version git-Paper-225 (MC: 1.18.2) (Implementing API version 1.18.2-R0.1-SNAPSHOT) (Git: 1d7a6a0)
You are running the latest version
Previous version: git-Paper-404 (MC: 1.17.1)

**Comment by Machine-Maker** [comment]
Yeah, I fixed this in the 1.18.2 update.
